ts-web-component/src/component/todo-item.ts

```diff
@@ -54,31 +57,44 @@ export default class TodoItem extends HTMLElement {
     return this._todoID;
   }
 
-  disconnectedCallback() {}
+  get text() {
+    return this.getAttribute("text") ?? "";
+  }
+
+  set text(v: string) {
+    this.setAttribute("text", v);
+  }
 
-  static get observedAttributes() {
-    return ["text", "checked"];
+  get completed() {
+    return this.hasAttribute("completed");
   }
 
-  attributeChangedCallback(_name: any, _oldValue: any, newValue: string) {
-    if (_name === "text") {
-      this._text = newValue;
-    } else if (_name === "checked") {
-      this._completed = newValue === "true";
+  set completed(v: boolean) {
+    if (v) {
+      this.setAttribute("completed", "");
+    } else {
+      this.removeAttribute("completed");
```

Comment:
This is called a boolean attribute in DOM speak. The presence means true and the absence means false